Bluepill-cli/Bluepill-cli/Simulator/SimulatorMonitor.m

```diff
@@ -135,6 +137,10 @@ - (void)onTestCaseFailedWithName:(NSString *)testName inClass:(NSString *)testCl
      [currentTime timeIntervalSinceDate:self.lastTestCaseStartDate],
      fullTestName];
 
+    // Save screenshot for failed test
+    NSString *screenshotName = [fullTestName stringByReplacingOccurrencesOfString:@"/" withString:@"_"];
+    [self.screenshotService saveScreenshotForFailedTestWithName:screenshotName];
```

Comment:
I'm not sure there is a difference - failed test are reported from various reasons, all inside BPTreeParser (correct?) and right after detecting failure we are passing this info to monitor..
This approach works from what I was testing locally